<span style='color:darkcyan'><font size="3">__Juan Sebastian Deslarzes__</font></span><br>
<span style='color:steelblue'><font size="3">__Chloé De Ancos__</font></span><br>
<span style='color:darkcyan'><font size="3">__Anaïs Burrus__</font></span><br>
<span style='color:darkcyan'><font size="3">__Margot Chapot__</font></span><br>
<span style='color:darkcyan'><font size="3">__Eva Sarlin__</font></span><br>

In [1]:
import matplotlib.pyplot as plt 
%matplotlib inline
import pandas as pd
import numpy as np
from skimage import io
from IPython import display
from bokeh.io import output_notebook, show
from bokeh.plotting import figure
output_notebook()
import seaborn as sns
sns.set(color_codes=True)

Loading BokehJS ...

In [2]:
from sklearn.base import BaseEstimator, TransformerMixin
from nltk.tokenize import sent_tokenize, word_tokenize
from nltk.stem import PorterStemmer
from nltk import word_tokenize
from nltk.corpus import stopwords
from nltk import pos_tag
from collections import Counter 
from sklearn.pipeline import Pipeline, FeatureUnion
from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer, TfidfTransformer
from sklearn.naive_bayes import MultinomialNB
from sklearn.model_selection import cross_val_score, KFold
from sklearn.metrics import classification_report
from sklearn.feature_extraction import DictVectorizer
from sklearn.preprocessing import FunctionTransformer
from sklearn.decomposition import NMF, LatentDirichletAllocation
import re
import nltk
from sklearn.datasets import load_files
nltk.download('stopwords')
import pickle
from nltk.stem import WordNetLemmatizer
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.feature_extraction.text import TfidfTransformer
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score

import string

[nltk_data] Downloading package stopwords to
[nltk_data]     /Users/evasarlin/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


<span style='color:mediumpurple'><font size="6">__Data Loading__</font></span><br>

In [86]:
sample_submission_url = 'https://raw.githubusercontent.com/esarlin/Challenge-2-CDAW/main/sitc-challenge2-2022/sample_submission.csv'
test_nolabel_url= 'https://raw.githubusercontent.com/esarlin/Challenge-2-CDAW/main/sitc-challenge2-2022/test_nolabel.csv'
train_url = 'https://raw.githubusercontent.com/esarlin/Challenge-2-CDAW/main/sitc-challenge2-2022/train.csv'

sample_submission = pd.read_csv(sample_submission_url)
test_nolabel = pd.read_csv(test_nolabel_url)
train = pd.read_csv(train_url)

In [33]:
from sklearn.linear_model import LogisticRegression

In [34]:
cvec=CountVectorizer()
X = train['statement']
y = train['label']
X_test = test_nolabel['statement']

In [35]:
X.head()

0    China is in the South China Sea and (building)...
1    With the resources it takes to execute just ov...
2    The (Wisconsin) governor has proposed tax give...
3    Says her representation of an ex-boyfriend who...
4    At protests in Wisconsin against proposed coll...
Name: statement, dtype: object

In [36]:
cvec=CountVectorizer(stop_words='english').fit(X)

In [37]:
df_train=pd.DataFrame(cvec.transform(X).todense(),columns=cvec.get_feature_names_out())

In [38]:
df_train.head()

00  000  000new  02  033  036  04  05  050  06  ...  zephyr  zero  zika  \
0   0    0       0   0    0    0   0   0    0   0  ...       0     0     0   
1   0    0       0   0    0    0   0   0    0   0  ...       0     0     0   
2   0    0       0   0    0    0   0   0    0   0  ...       0     0     0   
3   0    0       0   0    0    0   0   0    0   0  ...       0     0     0   
4   0    0       0   0    0    0   0   0    0   0  ...       0     0     0   

   zip  zombie  zombies  zone  zones  zuckerberg  ʺmore  
0    0       0        0     0      0           0      0  
1    0       0        0     0      0           0      0  
2    0       0        0     0      0           0      0  
3    0       0        0     0      0           0      0  
4    0       0        0     0      0           0      0  

[5 rows x 11151 columns]

In [39]:
df_test=pd.DataFrame(cvec.transform(X_test).todense(),columns=cvec.get_feature_names_out())

In [40]:
print (df_train.shape)

(8950, 11151)


In [41]:
print (df_test.shape)

(3836, 11151)


In [18]:
lr=LogisticRegression()

In [19]:
lr.fit(df_train,y)

/Users/evasarlin/opt/anaconda3/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [20]:
ypred=lr.predict(df_test)

In [42]:
X = train['subject']
y = train['label']
X_test = test_nolabel['subject']
cvec=CountVectorizer(stop_words='english').fit(X)
subject_train= pd.DataFrame(cvec.transform(X).todense(),columns=cvec.get_feature_names_out())
subject_test= pd.DataFrame(cvec.transform(X_test).todense(),columns=cvec.get_feature_names_out())

In [43]:
train2=pd.concat([df_train, subject_train], axis=1)
test2=pd.concat([df_test, subject_test], axis=1)
print (train2.shape)
print (test2.shape)

(8950, 11326)
(3836, 11326)


In [25]:
lr=LogisticRegression()
lr.fit(train2,y)

/Users/evasarlin/opt/anaconda3/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [26]:
ypred=lr.predict(test2)

In [ ]:
# adding speaker : 

In [55]:
speaktrain=train['speaker']

In [56]:
#speaktrain['speaker'] = speaktrain['speaker'].fillna('donald')
speaktrain = speaktrain.astype(str)

In [67]:
X = train['speaker']
y = train['label']
X_test = test_nolabel['speaker']
cvec=CountVectorizer(stop_words='english').fit(X)
speaker_train= pd.DataFrame(cvec.transform(X).todense(),columns=cvec.get_feature_names_out())
speaker_test= pd.DataFrame(cvec.transform(X_test).todense(),columns=cvec.get_feature_names_out())
train3=pd.concat([train2, speaker_train], axis=1)
test3=pd.concat([test2, speaker_test], axis=1)
print (train3.shape)
print (test3.shape)

(8950, 14347)
(3836, 14347)


In [58]:
lr=LogisticRegression()
lr.fit(train,y)
ypred=lr.predict(test)

/Users/evasarlin/opt/anaconda3/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [73]:
#train['speaker_job'].unique()
train['speaker_job'].isna().sum()

2481

In [75]:
(train['speaker_job'].str.lower()).value_counts()

u.s. senator                                    627
president                                       438
governor                                        368
president-elect                                 247
u.s. representative                             230
                                               ... 
santa fe mayor                                    1
chancellor and president, liberty university      1
organization                                      1
law professor and blogger                         1
promoting safe driving                            1
Name: speaker_job, Length: 1026, dtype: int64

In [78]:
train['speaker_job']= train['speaker_job'].fillna('u.s. senator')
test_nolabel['speaker_job']= test_nolabel['speaker_job'].fillna('u.s. senator')

In [79]:
X = train['speaker_job']
#X['speaker_job'] = X['speaker_job'].astype(str)
y = train['label']
X_test = test_nolabel['speaker_job'] #.astype(str)
cvec=CountVectorizer(stop_words='english').fit(X)
speakerjob_train= pd.DataFrame(cvec.transform(X).todense(),columns=cvec.get_feature_names_out())
speakerjob_test= pd.DataFrame(cvec.transform(X_test).todense(),columns=cvec.get_feature_names_out())
train4=pd.concat([train3, speakerjob_train], axis=1)
test4=pd.concat([test3, speakerjob_test], axis=1)
print (train4.shape)
print (test4.shape)
lr4=LogisticRegression()
lr4.fit(train4,y)
ypred=lr4.predict(test4)

(8950, 15402)
(3836, 15402)


/Users/evasarlin/opt/anaconda3/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [82]:
(train['state_info'].str.lower()).value_counts()

texas                879
florida              853
wisconsin            648
new york             579
illinois             487
                    ... 
atlanta                1
rhode island           1
the united states      1
new york               1
georgia                1
Name: state_info, Length: 77, dtype: int64

In [87]:
train['state_info']= train['state_info'].fillna('unknown')
test_nolabel['state_info']= test_nolabel['state_info'].fillna('unknown')

In [88]:
X = train['state_info']
#X['speaker_job'] = X['speaker_job'].astype(str)
y = train['label']
X_test = test_nolabel['state_info'] #.astype(str)
cvec=CountVectorizer(stop_words='english').fit(X)
state_train= pd.DataFrame(cvec.transform(X).todense(),columns=cvec.get_feature_names_out())
state_test= pd.DataFrame(cvec.transform(X_test).todense(),columns=cvec.get_feature_names_out())
train5=pd.concat([train4, state_train], axis=1)
test5=pd.concat([test4, state_test], axis=1)
print (train5.shape)
print (test5.shape)
lr5=LogisticRegression()
lr5.fit(train5,y)
ypred=lr5.predict(test5)

(8950, 15470)
(3836, 15470)


/Users/evasarlin/opt/anaconda3/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [90]:
X = train['party_affiliation']
#X['speaker_job'] = X['speaker_job'].astype(str)
y = train['label']
X_test = test_nolabel['party_affiliation'] #.astype(str)
cvec=CountVectorizer(stop_words='english').fit(X)
party_train= pd.DataFrame(cvec.transform(X).todense(),columns=cvec.get_feature_names_out())
party_test= pd.DataFrame(cvec.transform(X_test).todense(),columns=cvec.get_feature_names_out())
train5=pd.concat([train4, party_train], axis=1)
test5=pd.concat([test4, party_test], axis=1)
print (train5.shape)
print (test5.shape)
lr5=LogisticRegression()
lr5.fit(train5,y)
ypred=lr5.predict(test5)

(8950, 15435)
(3836, 15435)


/Users/evasarlin/opt/anaconda3/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [92]:
train5.head()

00  000  000new  02  033  036  04  05  050  06  ...  moderate  newsmaker  \
0   0    0       0   0    0    0   0   0    0   0  ...         0          0   
1   0    0       0   0    0    0   0   0    0   0  ...         0          0   
2   0    0       0   0    0    0   0   0    0   0  ...         0          0   
3   0    0       0   0    0    0   0   0    0   0  ...         0          0   
4   0    0       0   0    0    0   0   0    0   0  ...         0          0   

   ocean  official  organization  party  republican  state  talk  tea  
0      0         0             0      0           1      0     0    0  
1      0         0             0      0           0      0     0    0  
2      0         0             0      0           0      0     0    0  
3      0         0             0      0           0      0     0    0  
4      0         0             0      0           1      0     0    0  

[5 rows x 15435 columns]

<span style='color:mediumpurple'><font size="6">__Data Visualization__</font></span><br>

In [4]:
train.head()

id  label                                          statement  \
0  81f884c64a7      1  China is in the South China Sea and (building)...   
1  30c2723a188      0  With the resources it takes to execute just ov...   
2  6936b216e5d      0  The (Wisconsin) governor has proposed tax give...   
3  b5cd9195738      1  Says her representation of an ex-boyfriend who...   
4  84f8dac7737      0  At protests in Wisconsin against proposed coll...   

                                             subject  \
0                      china,foreign-policy,military   
1                                        health-care   
2           corporations,pundits,taxes,abc-news-week   
3  candidates-biography,children,ethics,families,...   
4                     health-care,labor,state-budget   

                      speaker            speaker_job        state_info  \
0                donald-trump        President-Elect          New York   
1                  chris-dodd           U.S. senator       Connecticut   
2               donna-brazile  Political commentator  Washington, D.C.   
3             rebecca-bradley                    NaN               NaN   
4  republican-party-wisconsin                    NaN         Wisconsin   

  party_affiliation  
0        republican  
1          democrat  
2          democrat  
3              none  
4        republican

In [5]:
print('Data train dimensions : ', train.shape)
print('Unique values for label : ', train['label'].unique())
print("Number of unique values for statement: ", train['statement'].unique().size)
print("Number of unique values for subject: ", (train['subject'].unique()).size)
print("Number of unique values for speaker: ", (train['speaker'].unique()).size)
print("Number of unique values for speaker job: ", (train['speaker_job'].unique()).size)
print("Number of unique values for state information: ", (train['state_info'].unique()).size)
print("Number of unique values for party affiliation: ", (train['party_affiliation'].unique()).size)

Data train dimensions :  (8950, 8)
Unique values for label :  [1 0]
Number of unique values for statement:  8939
Number of unique values for subject:  3409
Number of unique values for speaker:  2634
Number of unique values for speaker job:  1092
Number of unique values for state information:  80
Number of unique values for party affiliation:  24


<div class="alert alert-block alert-info">
    <span style='color:darkblue'><font size="4"><b>CONCLUSION DATA VISUALIZATION</b></font></span><br><br> <span style='color:black'> Donc en gros, on a un dataset de 8950 instances avec 8 colonnes. On peut y voir 7 features et 1 label/étiquette qui est la deuxième colonne "label". On doit rendre un un fichier submission avec les id et les label. L'objectif du challenge est donc de classifier en prédictant le label du fichier test_nolabel. On peut voir que la problème est le grand nombre de valeurs uniques pour une grande partie des attributs. Cela est du au fait que cette fois les valeurs des features sont du texte. L'objectif est donc de process ces valeurs textuelles pour pourvoir ensuite les uiliser avec des algo de machine learning. </span>
</div>

<span style='color:mediumpurple'><font size="6">__NLP algorithms (process les strings)__</font></span><br>

<span style='color:steelblue'><font size="3">__Avec le code du lab nlp (accuracy calculée sur le train)__</font></span><br>

In [48]:
X = train['statement'].values
y = train['label'].values


X_test = test_nolabel['statement'].values


In [6]:
# Generic Transformer 
class GenericTransformer(BaseEstimator, TransformerMixin):

    def transform(self, X, y=None):
        return do_something_to(X, self.vars)  # where the actual feature extraction happens

    def fit(self, X, y=None):
        return self  # used if the feature requires training, for example, clustering

In [7]:
class LexicalStats (BaseEstimator, TransformerMixin):
    """Extract lexical features from each document"""
    
    def number_sentences(self, doc):
        sentences = sent_tokenize(doc, language='english')
        return len(sentences)

    def fit(self, x, y=None):
        return self

    def transform(self, docs):
        return [{'length': len(doc),
                 'num_sentences': self.number_sentences(doc)}
                for doc in docs]

In [8]:
def custom_tokenizer(words):
    """Preprocessing tokens as seen in the lexical notebook"""
    tokens = word_tokenize(words.lower())
    porter = PorterStemmer()
    lemmas = [porter.stem(t) for t in tokens]
    stoplist = stopwords.words('english')
    lemmas_clean = [w for w in lemmas if w not in stoplist]
    punctuation = set(string.punctuation)
    lemmas_punct = [w for w in lemmas_clean if  w not in punctuation]
    return lemmas_punct

In [9]:
class PosStats(BaseEstimator, TransformerMixin):
    """Obtain number of tokens with POS categories"""

    def stats(self, doc):
        tokens = custom_tokenizer(doc)
        tagged = pos_tag(tokens, tagset='universal')
        counts = Counter(tag for word,tag in tagged)
        total = sum(counts.values())
        #copy tags so that we return always the same number of features
        pos_features = {'NOUN': 0, 'ADJ': 0, 'VERB': 0, 'ADV': 0, 'CONJ': 0, 
                        'ADP': 0, 'PRON':0, 'NUM': 0}
        
        pos_dic = dict((tag, float(count)/total) for tag,count in counts.items())
        for k in pos_dic:
            if k in pos_features:
                pos_features[k] = pos_dic[k]
        return pos_features
    
    def transform(self, docs, y=None):
        return [self.stats(doc) for doc in docs]
    
    def fit(self, docs, y=None):
        """Returns `self` unless something different happens in train and test"""
        return self

In [10]:
ngrams_featurizer = Pipeline([
  ('count_vectorizer',  CountVectorizer(ngram_range = (1, 3), encoding = 'ISO-8859-1', 
                                        tokenizer=custom_tokenizer)),
  ('tfidf_transformer', TfidfTransformer())
])

In [19]:
documents = []

stemmer = WordNetLemmatizer()

for sen in range(0, len(X)):
    # Remove all the special characters
    document = re.sub(r'\W', ' ', str(X[sen]))
    
    # remove all single characters
    document = re.sub(r'\s+[a-zA-Z]\s+', ' ', document)
    
    # Remove single characters from the start
    document = re.sub(r'\^[a-zA-Z]\s+', ' ', document) 
    
    # Substituting multiple spaces with single space
    document = re.sub(r'\s+', ' ', document, flags=re.I)
    
    # Removing prefixed 'b'
    document = re.sub(r'^b\s+', '', document)
    
    # Converting to Lowercase
    document = document.lower()
    
    # Lemmatization
    document = document.split()

    document = [stemmer.lemmatize(word) for word in document]
    document = ' '.join(document)
    
    documents.append(document)
print(documents)

['0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '2 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '1 0 0 0 0 0', '0 0 0 0 0 1', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 1 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '1 0 0 0 0 0', '0 0 0 0 0 0', '1 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 0 0 0 0 0', '0 1 0 0 0 0', '0 0 0 0 

In [12]:
def documents_generator(X):

    documents = []

    stemmer = WordNetLemmatizer()

    for sen in range(0, len(X)):
        # Remove all the special characters
        document = re.sub(r'\W', ' ', str(X[sen]))

        # remove all single characters
        document = re.sub(r'\s+[a-zA-Z]\s+', ' ', document)

        # Remove single characters from the start
        document = re.sub(r'\^[a-zA-Z]\s+', ' ', document) 

        # Substituting multiple spaces with single space
        document = re.sub(r'\s+', ' ', document, flags=re.I)

        # Removing prefixed 'b'
        document = re.sub(r'^b\s+', '', document)

        # Converting to Lowercase
        document = document.lower()

        # Lemmatization
        document = document.split()

        document = [stemmer.lemmatize(word) for word in document]
        document = ' '.join(document)
        
        documents.append(document)
    
    return documents

In [15]:
print(len(documents_generator(X)))

8950


In [17]:
print(len(documents))

8950


In [49]:
vectorizer = CountVectorizer(max_features=1500, min_df=5, max_df=0.7, stop_words=stopwords.words('english'))
docs1=documents_generator(X)
X = vectorizer.fit_transform(docs1).toarray()
tfidfconverter = TfidfTransformer()
X1 = tfidfconverter.fit_transform(X).toarray()
print(X1)
print(X1.shape)
DF1=pd.DataFrame(X1)
DF1.shape

[[0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 ...
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]]
(8950, 1500)


(8950, 1500)

In [50]:
X2 = train['subject'].values
docs2=documents_generator(X2)
vectorizer = CountVectorizer(max_features=1500, min_df=5, max_df=0.7, stop_words=stopwords.words('english'))
X = vectorizer.fit_transform(docs2).toarray()
tfidfconverter = TfidfTransformer()
X2df = tfidfconverter.fit_transform(X).toarray()
print(X2df)
print(X2df.shape)
DF2=pd.DataFrame(X2df)
DF2.shape

[[0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.44786643 ... 0.         0.         0.        ]
 ...
 [0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.        ]]
(8950, 160)


(8950, 160)

In [51]:
X3df = train['speaker'].values
docs3=documents_generator(X3df)
X = vectorizer.fit_transform(docs3).toarray()
X3 = tfidfconverter.fit_transform(X).toarray()
print(X3)
print(X3.shape)
DF3=pd.DataFrame(X3)
DF3.shape

[[0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 ...
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]]
(8950, 650)


(8950, 650)

In [ ]:
# test 

In [52]:
X1_test = test_nolabel['statement'].values
docs1=documents_generator(X1_test)
X = vectorizer.fit_transform(docs1).toarray()
X1test = tfidfconverter.fit_transform(X).toarray()
print(X1test)
print(X1test.shape)
df1T=pd.DataFrame(X1test)
df1T.shape

[[0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.        ]
 ...
 [0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.        ]
 [0.         0.         0.         ... 0.         0.         0.43098469]]
(3836, 1500)


(3836, 1500)

In [53]:
X2_test = test_nolabel['subject'].values
docs2=documents_generator(X2_test)
X = vectorizer.fit_transform(docs2).toarray()
X2test = tfidfconverter.fit_transform(X).toarray()
print(X2test)
print(X2test.shape)
df2T=pd.DataFrame(X2test)
print(df2T.shape)

[[0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 ...
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]]
(3836, 149)
(3836, 149)


In [54]:
X3testdf = test_nolabel['speaker'].values
docs3=documents_generator(X3df)
X = vectorizer.fit_transform(docs3).toarray()
X3test = tfidfconverter.fit_transform(X).toarray()
print(X3test)
print(X3test.shape)
df3T=pd.DataFrame(X3test)
print(df3T.shape)

[[0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 ...
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]]
(8950, 650)
(8950, 650)


(3836, 1)
(8950, 650)


In [42]:
resulttest = pd.concat([DF, df3T], axis=1)
resulttest.shape

(8950, 2150)

In [43]:
result = pd.concat([df1, df3], axis=1)
result.shape

(8950, 2150)

<span style='color:steelblue'><font size="3">__Avec un autre code__</font></span><br>

<span style='color:mediumpurple'><font size="6">__Feature Engineering__</font></span><br>

<span style='color:steelblue'><font size="5">__Statement Processing__</font></span><br>

<span style='color:steelblue'><font size="5">__Subject processing__</font></span><br>

<span style='color:steelblue'><font size="5">__Speaker features processing__</font></span><br>

<div class="alert alert-block alert-info">
    <span style='color:darkblue'><font size="4"><b>CONCLUSION DATA EXPLORING</b></font></span><br><br> <span style='color:black'> ici on écrit les conclusions, on peut même écrire en italique <i>voili </i>voilou. </span>
</div>

<span style='color:mediumpurple'><font size="6">__Machine Learning Algorithm__</font></span><br>

<span style='color:steelblue'><font size="5">__Ensemble Algos__</font></span><br>

<span style='color:steelblue'><font size="5">__Deep Learning__</font></span><br>

<span style='color:steelblue'><font size="5">__Genetic Algorithm__</font></span><br>

<div class="alert alert-block alert-info">
    <span style='color:darkblue'><font size="4"><b>CONCLUSION ALGORITHMS</b></font></span><br><br> <span style='color:black'> ici on écrit les conclusions, on peut même écrire en italique <i>voili </i>voilou. </span>
</div>

<span style='color:mediumpurple'><font size="6">__Prediction__</font></span><br>

<span style='color:mediumpurple'><font size="6">__Submission__</font></span><br>

In [91]:
id = test_nolabel['id']
dfsubmission = pd.DataFrame({'id': id, 'label': ypred}, columns=["id", "label"])
dfsubmission.to_csv('submissionlr6.csv', index=False, header=True)

<div class="alert alert-block alert-info">
    <span style='color:purple'><font size="4"><b>CONCLUSION GENERALE</b></font></span><br><br> <span style='color:black'> ici on écrit les conclusions, on peut même écrire en italique <i>voili </i>voilou. </span>
</div>